May have found a new issue:
```
Waiting for 65596413151
jq: parse error: Invalid numeric literal at line 1, column 10
Error: Process completed with exit code 5.
```

--- a/.github/workflows/integration-tests.yaml
+++ b/.github/workflows/integration-tests.yaml
@@ -247,9 +247,17 @@ jobs:
     steps:
       - id: matrix
         run: |
-          matrix='{"index": []}'
+          matrix='{"job": []}'
           for i in {1..30}; do
-            matrix="$(jq --arg index "${index}" '.index += [$index]' <<<"${matrix}")"
+            # Include a UUID to artificially increase the job name length
+            uuid="$(cat /proc/sys/kernel/random/uuid)"
+            matrix="$(
+              jq \
+              --arg index "${index}" \
+              --arg uuid "${uuid}" \
+              '.job += {index: $index, uuid: $uuid}' \
+              <<<"${matrix}"
+            )"
           done
 
           {
@@ -259,7 +267,7 @@ jobs:
           } | tee -a "$GITHUB_OUTPUT"
 
   exec-large:
-    name: Execute Large
+    name: Execute Large (${{ matrix.job.index }}, ${{ matrix.job.uuid }})
     needs: setup-large
     runs-on: ubuntu-latest
     permissions:
@@ -276,7 +284,8 @@ jobs:
         id: matrix-output
         with:
           yaml: |
-            index: ${{ matrix.index }}
+            index: ${{ matrix.job.index }}
+            uuid: ${{ matrix.job.uuid }}
           debug: true
 
   test-large:
@@ -289,7 +298,7 @@ jobs:
       - name: Expected JSON
         id: expected
         run: |
-          json="$(jq '.index | map({index: .})' <<<"${matrix}")"
+          json="$(jq '.job | map({.index, .uuid})' <<<"${matrix}")"
           {
             echo "json<<EOF"
             jq '.' <<<"${json}"

--- a/.github/workflows/integration-tests.yaml
+++ b/.github/workflows/integration-tests.yaml
@@ -235,6 +235,9 @@ jobs:
               }
             ]
 
+  # Workflow which contain a large number of jobs may run into "bash: Argument list too long"
+  # if we haven't reduced the job content internally. If this occurs failures may also occur
+  # on other integration tests.
   setup-large:
     name: Setup Large
     runs-on: ubuntu-latest
@@ -244,10 +247,9 @@ jobs:
     steps:
       - id: matrix
         run: |
-          matrix='{"uuid": []}'
+          matrix='{"index": []}'
           for i in {1..30}; do
-            uuid="$(cat /proc/sys/kernel/random/uuid)"
-            matrix="$(jq --arg uuid "${uuid}" '.uuid += [$uuid]' <<<"${matrix}")"
+            matrix="$(jq --arg index "${index}" '.index += [$index]' <<<"${matrix}")"
           done
 
           {
@@ -274,7 +276,7 @@ jobs:
         id: matrix-output
         with:
           yaml: |
-            uuid: ${{ matrix.uuid }}
+            index: ${{ matrix.index }}
           debug: true
 
   test-large:
@@ -287,7 +289,7 @@ jobs:
       - name: Expected JSON
         id: expected
         run: |
-          json="$(jq '.uuid | map({uuid: .})' <<<"${matrix}")"
+          json="$(jq '.index | map({index: .})' <<<"${matrix}")"
           {
             echo "json<<EOF"
             jq '.' <<<"${json}"

--- a/action.yaml
+++ b/action.yaml
@@ -165,6 +165,7 @@ runs:
       if: ${{ steps.matrix-jobs.outputs.num == strategy.job-total && steps.matrix-jobs.outputs.num-running > 1 }}
       shell: bash
       run: |
+        set -x
         # Wait for running matrix jobs
         start=$(date +%s)
 

